cluster.go

```diff
@@ -754,7 +754,9 @@ func (c *ClusterClient) process(ctx context.Context, cmd Cmder) error {
 		var err error
 
 		if attempt > 0 {
-			time.Sleep(c.retryBackoff(attempt))
+			if err := internal.Sleep(ctx, c.retryBackoff(attempt)); err != nil {
+				return err
```

Comment:
Does there need to be a `cmd.setErr(err)` here and elsewhere?

Reply:
It should be. It is just becomes hard to track when `cmd.setErr` should be called. I hope to address this in https://github.com/go-redis/redis/issues/1099